Basic_viewer/doc/Basic_viewer/CGAL/Graphic_storage.h

```diff
@@ -0,0 +1,87 @@
+
+namespace CGAL {
+
+/*!
+ \ingroup PkgBasicViewerClasses
+
+The class `Graphic_storage` stores points, segments, triangles, rays and lines. Elements can be added, possibly with associated colors. Non triangular faces can be directly added and are triangulated internally.
+
+\tparam BufferType the number type used for point coordinates: `float` by default.
+
+*/
+template <typename BufferType=float>
+class Graphic_storage {
+public:
+  /// adds the given point in the storage.
+  template <typename KPoint>
+  void add_point(const KPoint &p);
+
+  /// adds the given colored point in the storage.
+  template <typename KPoint>
+  void add_point(const KPoint &p, const CGAL::IO::Color &acolor);
+
+  /// adds the given segment in the storage.
+  template <typename KPoint>
+  void add_segment(const KPoint &p1, const KPoint &p2);
+
+  /// adds the given colored segment in the storage.
+  template <typename KPoint>
+  void add_segment(const KPoint &p1, const KPoint &p2,
+                   const CGAL::IO::Color &acolor);
+
+  /// adds the given ray in the storage: an half line starting from `p` and having `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_ray(const KPoint &p, const KVector &v);
+
+  /// adds the given colored ray in the storage: an half line starting from `p` and having `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_ray(const KPoint &p, const KVector &v,
+               const CGAL::IO::Color &acolor);
+
+  /// adds the given line in the storage, defined by `p` and `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_line(const KPoint &p, const KVector &v);
+
+  /// adds the given colored line in the storage, defined by `p` and `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_line(const KPoint &p, const KVector &v,
+                const CGAL::IO::Color &acolor);
+
+  /// starts a new face.
+  void face_begin();
+
+  /// starts a new colored face.
+  void face_begin(const CGAL::IO::Color &acolor);
+
+  /// return `true` iff a face is started.
+  bool is_a_face_started() const;
```

Comment:
Do we really need a function for that?   OpenGL3 does not have something similar.

Reply:
Yes we need ! Thanks to this function, an user can draw any polygonal face easily.

Reply:
OpenGL does without.   A nested `beginFace()` will give an error message instead.

Reply:
If I remove this code from basic viewer, I need to re-add it in each package that want to visualize non triangular faces (thus about 10 times or more).

Here such a code is factorized (and it should be also factorized with three demo...)